src/main/java/nus/climods/logic/parser/ParserUtil.java

```diff
@@ -88,4 +92,21 @@ public static Optional<String> parseFacultyCode(String input) {
         return Optional.of(input);
     }
 
+    /**
+     * Parse semester details according to user input
+     * @param input Input string from user containing semester details
+     * @return Optional of String representing semester number. Returns empty Optional if user input is not a valid
+     *      semester.
+     */
+    public static Optional<SemestersEnum> parseSemesterType(String input) {
+        List<String> acceptedSemesterType = Arrays.asList("S1", "S2", "ST1", "ST2");
+        String toCheck = input.trim().toUpperCase();
+
+        if (!acceptedSemesterType.contains(toCheck)) {
+            return Optional.empty();
+        }
+
+        BigDecimal index = new BigDecimal(acceptedSemesterType.indexOf(toCheck) + 1);
+        return Optional.of(SemestersEnum.fromValue(index));
+    }
```

Comment:
Maybe it will be better and easier for us if we maintain our own semester enum or we modify the existing `SemesterEnum` to keep track of identifiers in `acceptedSemesterType`.

This way we use `fromValue` to get a `SemesterEnum` we can use and replace all string type semester.

Reply:
Will edit accordingly, and update on modifications made

Reply:
Updated SemestersEnum such that the Enum will also check if input is valid or invalid. The method to check for the input is `fromValue`